h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -214,51 +218,56 @@ public void computeImpl() {
-
-      // Compute metalearner
-      if(metalearnerAlgoImpl != null) {
-        Key<Model> metalearnerKey = Key.<Model>make("metalearner_" + metalearnerAlgoSpec + "_" + _model._key);
-
-        Job metalearnerJob = new Job<>(metalearnerKey, ModelBuilder.javaName(metalearnerAlgoImpl.toString()),
-                "StackingEnsemble metalearner (" + metalearnerAlgoSpec + ")");
-        //Check if metalearner_params are passed in
-        boolean hasMetaLearnerParams = _model._parms._metalearner_parameters != null;
-        long metalearnerSeed = _model._parms._seed;
-
-        Metalearner metalearner = Metalearner.createInstance(metalearnerAlgoSpec);
-        metalearner.init(
-          levelOneTrainingFrame,
-          levelOneValidationFrame,
-          _model._parms._metalearner_parameters,
-          _model,
-          _job,
-          metalearnerKey,
-          metalearnerJob,
-          _parms,
-          hasMetaLearnerParams,
-          metalearnerSeed
-        );
-        metalearner.compute();
-      } else {
-        throw new H2OIllegalArgumentException("Invalid `metalearner_algorithm`. Passed in " + metalearnerAlgoSpec +
-            " but must be one of " + Arrays.toString(Metalearner.Algorithm.values()));
+        Metalearner.Algorithm metalearnerAlgoSpec = _model._parms._metalearner_algorithm;
+        Metalearner.Algorithm metalearnerAlgoImpl = Metalearner.getActualMetalearnerAlgo(metalearnerAlgoSpec);
+
+        // Compute metalearner
+        if (metalearnerAlgoImpl != null) {
+          Key<Model> metalearnerKey = Key.<Model>make("metalearner_" + metalearnerAlgoSpec + "_" + _model._key);
+
+          Job metalearnerJob = new Job<>(metalearnerKey, ModelBuilder.javaName(metalearnerAlgoImpl.toString()),
+                  "StackingEnsemble metalearner (" + metalearnerAlgoSpec + ")");
+          //Check if metalearner_params are passed in
+          boolean hasMetaLearnerParams = _model._parms._metalearner_parameters != null;
+          long metalearnerSeed = _model._parms._seed;
+
+          Metalearner metalearner = Metalearner.createInstance(metalearnerAlgoSpec);
+          metalearner.init(
+                  levelOneTrainingFrame,
+                  levelOneValidationFrame,
+                  _model._parms._metalearner_parameters,
+                  _model,
+                  _job,
+                  metalearnerKey,
+                  metalearnerJob,
+                  _parms,
+                  hasMetaLearnerParams,
+                  metalearnerSeed
+          );
+          metalearner.compute();
+        } else {
+          throw new H2OIllegalArgumentException("Invalid `metalearner_algorithm`. Passed in " + metalearnerAlgoSpec +
+                  " but must be one of " + Arrays.toString(Metalearner.Algorithm.values()));
+        }
+      } catch (Throwable t){
+        // If there is an exception during model training or validation, remove the model before exiting
+        _model.remove();
```

Comment:
Remove the model if there is ANY error/exception inside and then rethrow it. Preventing leaks.

Reply:
there should be a try-finally,

this is an example of a correct way of doing that: https://github.com/h2oai/h2o-3/blob/master/h2o-algos/src/main/java/hex/tree/SharedTree.java#L223

please note that the outer `Model model` is not set until we actually acquire the lock, otherwise we would try to unlock a Model that was not locked in the finally block

Reply:
IMHO the catch block shouldn't be there, no other Drivers have that as @seb-h2o pointed out

there should also not be any parameter validation, that should be done in init phase before the model training is already in progress, I am referring to line 264

Reply:
@michalkurka : yeah agree, there are already many `H2OIllegalArgumentException` all over the place in this class: I'd suggest to create a separate refactoring ticket though to make proper parameters validation for SE: `StackedEnsemble` needs to override `init` and do all validation there, and probably call `StackedEnsembleModel#checkAndInheritModelProperties` during this validation.

Reply:
I wanted to override init. I'll move the code according to the comments above. Makes sense. I'll do just the quickfix. @seb-h2o Please create the JIRA if you want to, I'll leave the refactoring to the AutoML team.

Reply:
https://0xdata.atlassian.net/browse/PUBDEV-6588

Reply:
@Pscheidl I don't understand the model removal in the finally block: it means it's gonna be removed whatever happens.
Did you mean `unlock` as @michalkurka  suggested?

Reply:
@seb-h2o Oh yes ! Of course ! Of course, I never meant to do that. Sorry and thank you for discovering it. Resolved in https://github.com/h2oai/h2o-3/pull/3592/commits/d2f6a14814817829fa52dd82733a2e6fc2e80d30

Reply:
@seb-h2o I actually decided to completely revert the changes in compute-impl (try-finally block was the only change). Are you ok with it ? It's not in the scope of this PR and it cause some tests to fail. I'd rather avoid it. My goal is to introduce some validation and leave the rest of the refactorings for later.